File: Assignment10/3)/MortgageTest.java

import java.util.Scanner;

public class MortgageTest {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        
        double loan = -1;
        while (loan <= 0) {
            System.out.print("Enter the loan amount (positive number): ");
            loan = scanner.nextDouble();
            if (loan <= 0) {
                System.out.println("Invalid input! Loan amount must be positive.");
            }
        }

     
        double rate = -1;
        while (rate <= 0) {
            System.out.print("Enter the annual interest rate (positive number): ");
            rate = scanner.nextDouble();
            if (rate <= 0) {
                System.out.println("Invalid input! Interest rate must be positive.");
            }
        }

      
        int years = -1;
        while (years <= 0) {
            System.out.print("Enter the number of years of the loan (positive number): ");
            years = scanner.nextInt();
            if (years <= 0) {
                System.out.println("Invalid input! Number of years must be positive.");
            }
        }

       
        Mortgage mortgage = new Mortgage(loan, rate, years);

       
        mortgage.displayLoanDetails();

       
        scanner.close();
    }
}

File: Assignment10/3)C++/MortgageTest.cpp

#include <iostream>
#include "Mortgage.h"  
using namespace std;

int main() {
    double loan;
    double rate;
    int years;

  
    do {
        cout << "Enter the loan amount (positive number): ";
        cin >> loan;
        if (loan <= 0) {
            cout << "Invalid input! Loan amount must be positive." << endl;
        }
    } while (loan <= 0);


    do {
        cout << "Enter the annual interest rate (positive number): ";
        cin >> rate;
        if (rate <= 0) {
            cout << "Invalid input! Interest rate must be positive." << endl;
        }
    } while (rate <= 0);

   
    do {
        cout << "Enter the number of years of the loan (positive number): ";
        cin >> years;
        if (years <= 0) {
            cout << "Invalid input! Number of years must be positive." << endl;
        }
    } while (years <= 0);

  
    Mortgage mortgage(loan, rate, years);

    
    mortgage.displayLoanDetails();

    return 0;
}
